jdaviz/configs/mosviz/helper.py

```diff
@@ -46,89 +55,133 @@ def _extend_world(self, spec1d, ext):
+            return
+
+        min_1d = self._scales1d.min
+        max_1d = self._scales1d.max
+
+        spec1d = table_viewer._selected_data["spectrum-viewer"]
+        extend_by = int(self.app.data_collection[spec1d]["World 0"].shape[0])
+        world = self._extend_world(spec1d, extend_by)
+
+        # Workaround for flipped data
+        min_world, max_world = (world[0], world[-1]) if not self._is_world_flipped()\
+            else (world[-1], world[0])
+
+        # Warn the user if they've panned far enough away from the data
+        # that the viewers will desync
+        if min_1d < min_world or max_1d > max_world:
+            self._show_panning_warning()
+            return
+
+        self._panning_warning_triggered = False
+
+        idx_min = float((np.abs(world - min_1d)).argmin()) - extend_by
+        idx_max = float((np.abs(world - max_1d)).argmin()) - extend_by
+
+        self._update_in_progress = True
+        with self._scales2d.hold_sync():
+            self._scales2d.min = idx_min
+            self._scales2d.max = idx_max
+
+        self._update_in_progress = False
+
+    def _update_spec1d_x_axis(self, _):
         # This assumes the two spectrum viewers have the same x-axis shape and
         # wavelength solution, which should always hold
-        if change['old'] is None:
-            pass
-        else:
-            name = change['name']
-            if name not in ['min', 'max']:
-                return
-            new_idx = int(np.around(change['new']))
-            spec1d = self.app.get_viewer('table-viewer')._selected_data["spectrum-viewer"]
-            extend_by = int(self.app.data_collection[spec1d]["World 0"].shape[0])
-            world = self._extend_world(spec1d, extend_by)
-
-            scales = self.app.get_viewer('spectrum-viewer').scales
-            old_val = getattr(scales['x'], name)
-
-            # Warn the user if they've panned far enough away from the data
-            # that the viewers will desync
-            try:
-                val = world[new_idx+extend_by]
-            except IndexError:
-                val = old_val
-                msg = "Warning: panning too far away from the data may desync \
+        table_viewer = self.app.get_viewer('table-viewer')
+
+        if self._update_in_progress or table_viewer.row_selection_in_progress:
+            return
+
+        min_2d = self._scales2d.min
+        max_2d = self._scales2d.max
+
+        spec1d = table_viewer._selected_data["spectrum-viewer"]
+        extend_by = int(self.app.data_collection[spec1d]["World 0"].shape[0])
+        world = self._extend_world(spec1d, extend_by)
+
+        idx_min = int(np.around(min_2d)) + extend_by
+        idx_max = int(np.around(max_2d)) + extend_by
+
+        # Warn the user if they've panned far enough away from the data
+        # that the viewers will desync
+        if not (0 <= idx_min < len(world) and 0 <= idx_max < len(world)):
```

Comment:
Why not use the same logic syntax as L79 above?

Reply:
I don't understand what you mean. (is it using `or` instead of `not ..and ... `? )

Reply:
Isn't it more consistent with above if you do something like this or did I miss something?
```python
max_world = len(world)
if idx_min < 0 or idx_max >= max_word:
    # ...
```

Reply:
Ah, yes, it would be, but because of the flipped data workaround `idx_min` can be bigger than `idx_max`. Now that I think of it, it's probably better to swap `min_2d` and `max_2d` at the top of the method based on `_is_world_flipped`, then use your suggested logic.

Reply:
Sadly, that idea falls apart when setting the axis at the end of the method, in which min and max should be flipped. I will add a comment  instead of changing the logic.

Reply:
Thanks for checking! Yes, a comment would help since "min" and "max" mean something and it is weird when they can be flipped at will in this method.